core/src/test/scala/org/bykn/bosatsu/EvaluationTest.scala

```diff
@@ -1251,4 +1251,25 @@ main = match Bar(a):
     }
   }
 
+  test("pattern example from pair to triple") {
+    runBosatsuTest(
+      List("""
+package A
+
+struct Pair(f, s)
+struct Trip(f, s, t)
+
+Trip(a, b, c) = match Pair(1, "two"):
+  Pair(f, s): Trip(3, s, f)
```

Comment:
this seems like your example, and the evaluation codes seems to execute it correctly. What do you think @snoble am I misunderstanding Normalization?

Reply:
Oops, I need to clean up my notifications so I see these things. Looking

Reply:
Yeah, this definitely looks like what Normalization _should_ be doing. It could be the constructor code or the pattern matching code (in Normalization)